source/opt/ssa_rewrite_pass.cpp

```diff
@@ -681,6 +683,12 @@ Pass::Status SSARewriter::RewriteFunctionIntoSSA(Function* fp) {
             << fp->PrettyPrint(0) << "\n";
 #endif
 
+#if SSA_REWRITE_DEBUGGING_LEVEL > 1 || !defined(NDEBUG)
+  for (auto& bb : (*fp)) {
+    bb.SetUserData(nullptr);
+  }
```

Comment:
In a release build, we will leave many dangling pointers. I don't think this is a good idea. Debug these error will be hard especially if the error is only in the release build. I know it is time consuming, but we should be doing this all of the time.

Reply:
We should probably add something to the debug builds to check that all of the user data is null after each pass. Probably something like the call to `IsConsistent`.

https://github.com/KhronosGroup/SPIRV-Tools/blob/cc07e5303f1613d8577c2ff61995de39ccc2effd/source/opt/pass.cpp#L43